overlay/lib/listen.js

```diff
@@ -93,21 +94,42 @@
     }
 
     sort(key, target) {
-      let d = (('+-'.indexOf(key[0]))+1 || 1) * 2 - 3
-      let k = SORTABLE[key.substr('+-'.indexOf(key[0]) >= 0)]
-      ;(target || this.data).sort((a, b) => (pFloat(a[k]) - pFloat(b[k])) * d)
+      let order = (('+-'.indexOf(key[0]))+1 || 1) * 2 - 3 // 1:asc, -1:desc
+      let sort_by = SORTABLE[key.substr('+-'.indexOf(key[0]) >= 0)]
 
+      if (typeof sort_by == "string") {
+        (target || this.data).sort((a, b) => (pFloat(a[sort_by]) - pFloat(b[sort_by])) * order)
+      } else if (typeof sort_by == "function") {
+        (target || this.data).sort((a, b) => (sort_by(a) - sort_by(b)) * order)
+      }
       if(target) return target
     }
 
+    // Calculate additional healing values and store them in the Combatant or Encounter
+    calculateHealingValues(data) {
+      let rhealing_effective = 0;
+      let rabsorb_healing = 0;
+      for (let i in data.Combatant) {
+        let player = data.Combatant[i];
+        let effective = parseInt(player.healed) - parseInt(player.overHeal);
+        // Inject the calculated effective healing into the player data, while we're at it
+        player.effective_healing = effective;
```

Comment:
IMO actually, the term 'effective healing' should not be a main metric to compare between healers within party. I'm not saying including absorbs & shields is bad, but excluding overheals are.

Let's imagine a full party with WHM & AST that both healers are never talked & both doing almost max effort they can normally do.
On next raidwide AoE, someone needed to heal this or we all gonna die. WHM prepares `Plenary Indulgence` + `Afflatus Rapture` both has no casting, AST prepares Huge `Stellar Detonation` + `Horoscope Helios` also almost instant both are.
So the AoE came, that two healers activates their skills in same time. who achieved lower Overheals, who made more Effective Heals? if WHM's 100ms faster, can we blame AST as 'they did nothing'? no.

(I made WHM & AST as example as they're what I main, but if we think of SGE, my friends are shouting 'I should _do_ a heal to fill my MP', despite I can't know their truth as I never been SGE.)

Overheals are problem of Negotiation in most cases, doesn't mean they're underskilled.
and as Effective Healing excludes Overheal completely, so we need to think more on this topic.

Reply:
I believe the overhealing generated for this kind of specific moments would even out over the course of a fight, but I haven't healed in raids so I might be wrong.

I agree that overhealing is a normal part of healing, and you can't compare healers based this metric. Likewise,hps alone is not a good indicator of healer performance.
I think it's up to the user to understand and combine several metrics to evaluate performance (And in that case, more available metrics is better).
Plus, we already provide the overhealing %, so it's just a quick math for the user to calculate effective healing himself.

I wouldn't add it to the default columns, but I don't see the harm in having them available if someone wants them.

Reply:
Providing effective/absorbed value is absolutely does not matter, and it should be, but using it as single main metric (so removing instead of adding overheal) would be problematic as it hides before decision.

Actual solution should be create 'unified healing gauge' as like other overlays do, but this is quite a headache for current codebase...
![image](https://user-images.githubusercontent.com/5380174/156889278-007c3cd7-3f37-4c0e-95f3-449cd9b53fa5.png)
We would just add this for now & replace with this somewhen soon as possible.